pkg/types/message_envelope.go

```diff
@@ -164,3 +188,53 @@ func fromContext(ctx context.Context, key string) string {
 	}
 	return hdr
 }
+
+func ConvertPayloadToByteArray(payload any) ([]byte, error) {
```

Comment:
```suggestion
func ConvertMsgPayloadToByteArray(msg MessageEnvelope) ([]byte, error) {
```

check the content type.  if it's CBOR, use cbor.Marshal instead of json.Marshal